### ISSUE TITLE
[FR] Database cell link another page inline

### ISSUE THREAD

**Issue by annieappflowy** [AppFlowy-IO/AppFlowy#2196]
### Description

Support linking another page **inline** within a paragraph

Enable a few ways to trigger this function:
- [x] Type `@`, then start typing the name of the page you want to link. Select the page you'd like to link in the dropdown that appears.
- [x] #3541
- [x] Type `+`, then start typing the name of the page you want to link. Select the page you'd like to link in the dropdown that appears.
- [x] #2256

When the function is triggered and no input yet, the dropdown menu should show at most five latest viewed pages. As the user is typing the page name, the dropdown menu dynamically returns relevant results. If no results, then show "No results" in the menu.

Pages include `Document`, `Board`, `Grid`, `Calendar`, and row records in a database.
This feature will be expanded to support `Mention a block` in the future.
This feature is related to another feature `Backlinks` https://github.com/AppFlowy-IO/AppFlowy/issues/2214

### Impact

Users who need to mention other pages inline within a paragraph

**Comment by JRS296** [comment]
Has this feature been implemented yet? Or can this be worked on? @annieappflowy

**Comment by LucasXu0** [comment]
Hi, @JRS296. You can start by implementing this feature: 'Use the slash command /mention to trigger Mention a page'. Also, please let me know how you plan to implement it for the issue I assigned to you.

**Comment by JRS296** [comment]
@LucasXu0 I'll have a look at the code base, and I shall come up with a plan for implementation. Will update the comment thread on #2256 shortly.

**Comment by annieappflowy** [comment]
Status?

**Comment by JRS296** [comment]
> Status?

It's still incomplete, far from ready. Someone else can take over if it needs to finished quickly.

**Comment by rizwan3395** [comment]
I have started working on it @LucasXu0. Please assign it to me

**Comment by annieappflowy** [comment]
Should continue to support other ways described in this issue to link another page inline
cc: @LucasXu0

**Comment by sanketss84** [comment]
is there a way to link internal pages in this app ?

**Comment by annieappflowy** [comment]
> is there a way to link internal pages in this app ?

Have you tried `@`

**Comment by sanketss84** [comment]
my bad, thanks for your reply.

**Comment by annieappflowy** [comment]
No worries! I should take the blame because we don't make it intuitive enough and discoverable.

**Comment by Xazin** [comment]
> > > is there a way to link internal pages in this app ?
> > 
> > 
> > Have you tried `@`
> 
> Thx, I discover it only by reading this issue. I was saying to myself: this feature should be somewhere, I was trying and searching, but nothing until I arrived here.

We really want to improve our feature visibility, if I may ask, what were the first places you looked to see if you could find this information? And where do you expect to be able to find it?

**Comment by DonPiotr** [comment]
> > > > is there a way to link internal pages in this app ?
> > > 
> > > 
> > > Have you tried `@`
> > 
> > 
> > Thx, I discover it only by reading this issue. I was saying to myself: this feature should be somewhere, I was trying and searching, but nothing until I arrived here.
> 
> We really want to improve our feature visibility, if I may ask, what were the first places you looked to see if you could find this information? And where do you expect to be able to find it?

First I was trying some markdown-like syntax: [, [[, {, {{ , but nothing.
Then I tried slash menu, but also here nothing.
After I went to Getting started page, but even here no mention.
Then I found this issue.

**Comment by tagd** [comment]
Hi @annieappflowy
You mentioned in your issue "Pages include . . . row records in a database", I've tried all the syntaxes listed but none of them return the database pages, I've tried in both the database itself, database pages and standalone pages to no avail.  Is this feature complete or is there another FR for it? Thanks

The way I'd like to use this is as the below where I can link from a table of projects or similar to the current task for that project(in another database), though just being able to link in the page for that row would allow me to switch from notion for project management.
![image](https://github.com/AppFlowy-IO/AppFlowy/assets/45215866/b659f20e-4114-46d9-bbe1-1d121ddbb398)

**Comment by rodeleeuw26061972** [comment]
Inline links to pages don't open in the mobile version. Any progress on this issue?

**Comment by Xazin** [comment]
> Internal links to pages don't open in the mobile version. Any progress on this issue?

I wasn't able to reproduce on version 0.5.6 on iPhone (iOS 17.4.1).

Are you on the latest version on mobile?

**Comment by ivishalgandhi** [comment]
Does implementation of page reference using [[page]] works in grid layout inside a cell ?

**Comment by haron-IV** [comment]
> Does implementation of page reference using [[page]] works in grid layout inside a cell ?

Just testes it now and it doesn't.

**Comment by rileyhawk1417** [comment]
As of 0.8.1 you can link grids using the relation property.

![Image](https://github.com/user-attachments/assets/9861ef89-2c47-4ac0-b1fc-7fd415a3907c)
Screenshot showing related property

![Image](https://github.com/user-attachments/assets/62cfa71f-dd2d-4237-8899-32839a867d80)
Screenshot selecting a row to link with.

**Comment by AntoineGlacet** [comment]
Hello,
just stumble on this issue after a google search.
Linking pages from a database is really an important feature.
After using Appflowy for a few days, it would totally be the #1 feature I would appreciate.

For example, I organize my ideas with one sub-page per idea and I would love to use a DB to sort them and manage them (prioritizing, scheduling, etc). If I could link DB and page it would be great.
Basically linking DB and pages, which are the 2 main logical entities of appflowy would unlock a lot of potential.
